### ISSUE TITLE
Add support for drone.io

### ISSUE THREAD

**Issue by cdvv7788** [pydanny/cookiecutter-django#2082]
## Description
Drone.io support built-in
[//]: # (What's it you're proposing? How should it be implemented?)
Add a .drone.yml file that allows projects to be tested/built using drone.io

## Rationale

[//]: # (Why should this feature be implemented?)
There are options for several areas of the project. Why not add some more flavor to the CI/CD options?
[Drone ](https://github.com/drone/drone) is very nice, from what I have tested. Also, it is open source, unlike most solutions out there. This makes it a candidate for baseline support.

## Use case(s) / visualization(s)

[//]: # ("Better to see something once than to hear about it a thousand times.")
Same as travis, actually. Adding some more options to people who uses other tools. This could be selected at install time.

**Comment by browniebroke** [comment]
Sure, feel free to send a pull request. I've heard of it, but I have no experience with this CI, I hope others may help to get the change reviewed... It sounds like it would be all in one config file so I don't mind having this.

**Comment by cdvv7788** [comment]
Yes, it just requires adding a single file to the repo. The installer should ask for the option (travis, drone)? Or just add them both?

**Comment by browniebroke** [comment]
Currently it asks for Travis Y/N so this could be altered to choose a CI: Travis/Drone/None

**Comment by step21** [comment]
or travis/drone/gitlab...

**Comment by shywn-mrk** [comment]
Since the project is using GitHub Actions now is this still relevant?

**Comment by step21** [comment]
> Since the project is using GitHub Actions now is this still relevant?

Probably, since ppl might prefer to use existing ci services instead of GA.

**Comment by browniebroke** [comment]
This is still relavant, the template uses GitHub action, but the generated project might be using a totally different CI. We support a few at the moment:

https://github.com/cookiecutter/cookiecutter-django/blob/3bca38c78d77ecdd9b1ae9af52b373c4277a9c1a/cookiecutter.json#L57-L62

I'm open to add this to the existing option, assuming it's still a single file. Caveat: I've never used drone, so I can't do it... @cdvv7788 @shywn-mrk @step21 are you interested to contribute?

**Comment by shywn-mrk** [comment]
@browniebroke Yes I'm willing to contribute. Where can I start?

**Comment by browniebroke** [comment]
I would recommend reading our [contributing guide](https://cookiecutter-django.readthedocs.io/en/latest/contributing.html) which explains the specifics of this project. 

You can also check the GitHub docs themselves you need some help to use GitHublike for [creating a fork](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks) or [sending a pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork).

Here are the high level changes:

1. Add the drone config files in the `{{cookiecutter.project_slug}}` directory
2. Add a new option for drone CI in `cookiecutter.json`:
    https://github.com/cookiecutter/cookiecutter-django/blob/8d64f1749461310b04a2b512184bb0623fb22ba0/cookiecutter.json#L42
3. Add the new option to the `SUPPORTED_COMBINATIONS` in tests: 
    https://github.com/cookiecutter/cookiecutter-django/blob/8d64f1749461310b04a2b512184bb0623fb22ba0/tests/test_cookiecutter_generation.py#L45
4. Make sure the drone config file(s) is/are deleted if another option is selected. We usually do that in the [post-generation hooks](https://github.com/cookiecutter/cookiecutter-django/blob/master/hooks/post_gen_project.py)
5. Update documentation

Hope that helps
